### PR TITLE
Importance of Global phase emphasized in the Conditional Flip Task of Grover's Algorithm

### PR DESCRIPTION
**Pull request by Manvi-Agrawal**
PR for the following issue - https://github.com/microsoft/QuantumKatas/issues/491

### CODE REVIEW

File: GroversAlgorithm/GroversAlgorithm.ipynb [hunk 1]
```diff
@@ -270,6 +306,15 @@
     "    You need to modify it to add the $-1$ phase to only the $|0...00\\rangle$ state instead.\n",
     "    <br/><br/>\n",
     "    Alternatively, you can use the same trick as in the oracle converter task.<br>\n",
+    "</details>\n",
+    "<br>\n",
+    "\n",
+    "<details>\n",
+    "  <summary><b>IMPORTANT : The test will pass but its not the correct implementation. For More details click here</b></summary>\n",
```
Comment (tcNickolas) [inline GroversAlgorithm/GroversAlgorithm.ipynb:313]: Looks like this hint comes from the version of #505 before the latest version? Same comment as in #505, let's add this to hint 1. 

Also, I would make the information in hint 3 that you added in the latest version of that PR a lot shorter - the information about eigenvalues is not necessary and somewhat confusing, it's enough to point the learner to the R operation.

**Review by tcNickolas — CHANGES_REQUESTED** [review]
I made a pass based on both this PR and #505 - looks like some of the files came from the latest version of it and some from a previous one, so I tried to combine the best of both.

Thank you!

File: GroversAlgorithm/GroversAlgorithm.ipynb [hunk 2]
```diff
@@ -34,6 +34,42 @@
     "harder ones are marked with asterisks."
    ]
   },
+  {
```
[the remark this answers is not in the document]
Comment (Manvi-Agrawal) [reply GroversAlgorithm/GroversAlgorithm.ipynb]: Done

File: GroversAlgorithm/Tasks.qs [hunk 3]
```diff
@@ -135,11 +135,31 @@ namespace Quantum.Kata.GroversAlgorithm {
     // Note: This operation implements operator 2|0...0⟩⟨0...0| - I.
     operation ConditionalPhaseFlip (register : Qubit[]) : Unit is Adj {
     
-        // Hint 1: Note that quantum states are defined up to a global phase.
-        // Thus the state obtained as a result of this operation is the same
-        // as the state obtained by flipping the sign of only the |0...0⟩ state.
+        // Hint 1: Let's do instead I - 2|0...0⟩⟨0...0|, i.e. flipping the phase of only  |0...0⟩.
```
[the remark this answers is not in the document]
Comment (Manvi-Agrawal) [reply GroversAlgorithm/Tasks.qs]: Done.

File: GroversAlgorithm/Tasks.qs [hunk 4]
```diff
@@ -135,11 +135,31 @@ namespace Quantum.Kata.GroversAlgorithm {
...
+        // is used as a building block for other algorithms, in particular quantum counting algorithm,
+        // which converts the global phase difference into the relative phase,
+        // this can cause very interesting bugs. See an extended discussion in this Quantum Computing SE question.
+        // Link : https://quantumcomputing.stackexchange.com/questions/5973/counting-in-q-number-of-solutions/6446#6446
+        // However, the effect of extra global phase introduced in the circuit can be nullified by adding back
+        // global phase of  −1  back to the circuit
+        
             
         // Hint 2: You can use the same trick as in the oracle converter task.
+        // Alternatively, consider the Controlled Z gate, applied with most of the qubits as control and the last qubit as target.
```
[the remark this answers is not in the document]
Comment (Manvi-Agrawal) [reply GroversAlgorithm/Tasks.qs:151]: Done

File: GroversAlgorithm/GroversAlgorithm.ipynb (hunk 1 above)
Comment (Manvi-Agrawal) [reply GroversAlgorithm/GroversAlgorithm.ipynb]: Done

**Comment by Manvi-Agrawal** [comment]
Hi, @tcNickolas. I have made the changes as requested by you. Please have a look and lemme know if any further changes are required.

**Comment by jainvasu631** [comment]
@Manvi-Agrawal In the title you mention Deutsch Josza Algorithm while the previous issue #505 and the files changed are all for Grover's algorithm. 
Does the PR title need to be amended or am I missing something?

**Comment by Manvi-Agrawal** [comment]
Nice catch @jainvasu631. The changes are for Grover's Algorithm. I have ammended the PR title now to avoid any future confusions.

**Review by tcNickolas — APPROVED** [review]
Looks good - I did a bit of cleanup for clarity and the right level of mysteriousness, but not a lot.

Thank you!

**Pull request merged**